### DISCUSSION TITLE
Case-insensitive trigger

### DISCUSSION THREAD

**Discussion by mkjaxxo** [espanso/espanso#1245] [Feature requests and Ideas]
Hi @federico-terzi, 

case-insensitive triggers would be a great feature.

Example:

  - trigger: "wordpress"
    replace: "WordPress"
    case-insensitive: true

Result: All cases "wordpress", "Wordpress", "WORDPRESS", "WordpresS", ... would result in the replacement "WordPress".

**Comment by federico-terzi** [comment]
Hey @mkjaxxo,

Thank you for the feedback! Sounds very useful, I'll add it to the roadmap!

Cheers :)

**Comment by smeech** [comment]
An old discussion.
For reference, the current implementation supports:
```yml
  - regex: (?i)wordpress
    replace: WordPress
```

**Reply by SantiagoCarranza1980** [reply]
This should be in the docs as an example berfore sending you down the rabbit hole of regex!
Thank you for the tip!

**Reply by smeech** [reply]
Once you get into Regex, there are so many possibilities that it would be difficult to choose which to document, however! I can see why Federico chose to focus on wildcards and capturing groups.

**Comment by ivan-paleo** [comment]
Unfortunately, the `regex` solution cannot be combined with `word: true`. 
But I guess it should be possible to specify the regex so that it replaces only whole words; I just don't know how. Could someone help me?

**Reply by joh6nn** [reply]
`\b` marks a word boundary in regexes; you can use it as follows:
- `\bexample` match any word that begins with "example"
- `example\b` match any word that ends with example
- `\bexample\b` match only example


You can experiment with it at [regex101.com](https://regex101.com/r/iVSSa9/1)

Hope that helps

**Reply by ivan-paleo** [reply]
Thanks a lot!

**Reply by ivan-paleo** [reply]
Unfortunately, the last `\b` does not work: 
For example:
```
matches:
  - regex: \bexample\b
    replace: "test"
``` 
- `Xexample` doesn't get changed as expected
-  `exampleX` gets changed to `test` even before I get the chance to type `X`

Any idea what's wrong? Me, regex or Espanso?

**Reply by smeech** [reply]
Confirmed here, but the same in https://regex101.com/r/CpDYCb/1 and elsewhere, so it's probably correct and one would need to find a way around it. Dynamically, `\b` matches the end of a string, _with or without_ a character afterwards, so the trigger is invoked.
```yml
  - regex: (?i)\bexample\W
    replace: Test
```
is a better alternative, and closer to the `word: true` method, although it doesn't preserve the final (non-word) character.
```yml
  - regex: (?i)\bexample(?P<w>\W)
    replace: Test{{w}}
```
does.

**Reply by ivan-paleo** [reply]
Wonderful, thanks @smeech !
